django/core/checks/urls.py

```diff
@@ -44,9 +46,18 @@ def get_warning_for_invalid_pattern(pattern):
     describe_pattern() cannot be used here, because we cannot rely on the
     urlpattern having regex or name attributes.
     """
+    if isinstance(pattern, six.string_types):
+        hint = "Try removing the string {!r}. The list of urlpatterns should " \
```

Comment:
Good point. I used `__repr__` for the warning message, because we don't know what sort of object it is. But here we know it's a string, so we don't need to use `__repr__`. I've fixed that, and made the other changes you suggested.